### PR TITLE
Add IsCollectible property to Assembly and necessary backing functions

### PR DESCRIPTION
**Pull request by josalem**
Expose API to check whether the referenced `Assembly` is collectible, i.e., has been loaded into an Assembly Load Context that has been marked as Collectible.

CC - @sergiy-k

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs [hunk 1]
```diff
@@ -201,6 +201,20 @@ public override IEnumerable<TypeInfo> DefinedTypes
...
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool GetIsCollectible(RuntimeAssembly assembly);
+
+        public override bool IsCollectible
+        {
+            get
+            {
+                bool isCollectible = false;
+                isCollectible = GetIsCollectible(GetNativeHandle());
```
[the remark this answers is not in the document]
Comment (janvorli) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: Or even make it a one-liner like we do in corefx:
```C#
public override bool IsCollectible => GetIsCollectible(GetNativeHandle());
```

**Comment by jkotas** [comment]
Note that you will also need to do a matching change in CoreFX to expose this property in the reference assemblies and add tests.

**Comment by josalem** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

**Comment by josalem** [comment]
@dotnet-bot test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness please

**Pull request merged**